mon-pix/app/utils/proposals-as-blocks.js

```diff
@@ -1,121 +1,99 @@
 import isEmpty from 'lodash/isEmpty';
 
-function parseInput(isInput, input) {
-  let block;
-
-  switch (input) {
-    case '${':
-      isInput = true;
-      break;
-    case undefined:
-      isInput = false;
-      break;
-    case '':
-      break;
-    default:
-      if (isInput) {
-        block = new ResponseBlock({ input });
-      }
-      else {
-        block = new ResponseBlock({ text: input });
-      }
-  }
-
-  return { isInput, block };
-}
-
+const MINIMUM_SIZE_FOR_LABEL = 6;
 function stringHasPlaceholder(input) {
   return 1 <= input.indexOf('#');
 }
 
-function isLastElement(currentIdx, elements) {
-  return currentIdx === (elements.length - 1);
+function stringHasAriaLabel(input) {
+  return 1 <= input.indexOf('§');
 }
 
-function getLastLine(lines) {
-  return lines[lines.length - 1];
+function _isInput(block) {
+  return block.includes('${');
 }
 
-function isAriaLabelNeededForInputs(lines) {
-  const lastLine = getLastLine(lines);
-  const inputStartRegex = /\${/g;
-  const lastLineInputs = lastLine.match(inputStartRegex, '');
-  const hasMoreThanOneInputField = lastLineInputs && lastLineInputs.length > 1;
-  const inputRegex = /\s*(\${.+?})|-| /g; //regex that remove spaces, -, ${} => should return only letters
-  const lastLineWithoutInput = lastLine.replace(inputRegex, '');
-  const blocks = lastLine.split(/\s*(\${)|}\s*/);
-  const hasNoTextBeforeInput = !(blocks && blocks[0].replace(inputRegex, '').length > 0);
+function buildLineFrom(textBlock, challengeResponseTemplate) {
+  const isInput = _isInput(textBlock);
+  const block = textBlock;
 
-  if (hasMoreThanOneInputField || hasNoTextBeforeInput) {
-    return true;
-  }
+  if (isInput) {
+    challengeResponseTemplate.incrementInputCount();
+    const blockToTemplate = new InputBlock({ input: block, inputIndex: challengeResponseTemplate.inputCount });
+    blockToTemplate.attachInputAndPlaceholderIfExist();
+    challengeResponseTemplate.add(blockToTemplate);
 
-  return lastLineWithoutInput.length === 0;
+  } else {
+    const blockToTemplate = new TextBlock({ text: block });
+    challengeResponseTemplate.add(blockToTemplate);
+  }
 }
 
-function buildLineFrom(blocks, ariaLabelNeeded, challengeResponseTemplate) {
-  let previousBlockText = '';
-  let isInput = false;
+class TextBlock {
 
-  for (let blockIdx = 0; blockIdx < blocks.length; blockIdx += 1) {
-    const input = parseInput(isInput, blocks[blockIdx]);
-    const block = input.block;
+  constructor({ text }) {
+    if (text.substring(0, 1) === '-') {
+      text = '\n' + text;
+    }
```

Comment:
Suggestio pour que cela fonctionne aussi avec les listes numérotées:

```suggestion
    const firstCharacter = text.substring(0, 1);
    const secondCharacter = text.substring(1, 2);
    const isListItem = firstCharacter === '-' || Number.isInteger(Number(firstCharacter)) && secondCharacter=== '.';
    if (isListItem) {
      text = '\n' + text;
    }
```

Reply:
```suggestion
    const firstTextBeforeSpace = text.split(' ')[0];
    const containsOnlyNumberAndDot = (text) => { 
       for (let i = 0; i < text.length; i++) {
         const isNumber = Number.isInteger(Number(text[i]));
         const isDot = text[i] === '.';
         if (!isNumber || !isDot) return false;
       }
       return true;
    }
    const isListItem = firstTextBeforeSpace === '-'
        || Number.isInteger(Number(firstCharacter)) && secondCharacter === '.';
    if (isListItem) {
      text = '\n' + text;
    }
```